docs.feldera.com/docs/pipelines/modifying.md

```diff
@@ -0,0 +1,436 @@
+store, created using the [S3 sync](https://docs.feldera.com/pipelines/checkpoint-sync) feature. This checkpoint may have been created by an older
+version of the same pipeline or even by a different pipeline.  This situation is handled in the
+same way: the pipeline computes a diff between the current and the checkpointed query plan, bootstraps
+any new and modified views and clears any tables that are not part of the checkpoint.
+
+### Caveat 3: Views that have not been modified may require bootstrapping
+
+The Feldera SQL compiler performs some global optimizations across multiple views. Modifying or
+removing one of the views can affect these optimizations, potentially changing the query plan
+for other views, not directly affected by the change.
+
+### Limitation 1: Tables must be materialized
+
+Bootstrapping relies on existing pipeline state to initialize new and modified views.
+[Materializing](/sql/materialized) all tables in the program by declaring them with `'materialized' = 'true'`
+ensures that the pipeline can be bootstrapped after arbitrary changes from this materialized
+state.
+
+A program where not all tables are materialized may fail to start from a checkpoint and require
+clearing the pipeline's state and restarting the pipeline from scratch instead.
+
+### Limitation 2: Bootstrapping does not work with `LATENESS`
+
+Programs with [`LATENESS` annotations](/tutorials/time-series) currently cannot be bootstrapped. These annotations are used
+by the query engine to discard state that is not used to evaluate any of the views in the program.
+However, the modified program may contain a different set of queries that require the discarded state.
+
+We currently don't have the analysis needed to detect such changes and instead take the more coarse-grained
+approach of simply rejecting changes to pipelines with `LATENESS`. Such pipelines must be
+backfilled from scratch after any change by clearing their state before restarting.
+
+### Limitation 3: Table evolution is not yet supported
+
+Adding, removing or renaming a table column will clear the table's state and require the table
+to be ingested from scratch.
+
+## API
+
+The Feldera REST API contains several elements that support the bootstrapping feature described in this document:
+
+* The `bootstrap_policy` argument to the [`/start` endpoint](https://docs.feldera.com/api/stop-the-pipeline-asynchronously-by-updating-the-desired-state). This argument specifies how the pipeline should
+  behave when the checkpointed version of the program differs from the current version. The supported values are
+  `await_approval`, `allow`, and `reject`. The default value is `await_approval`. See the
+  [Restart the pipeline](#3-restart-the-pipeline) section above for details.
+
+* Two new runtime states reported by the [pipeline status endpoint](https://docs.feldera.com/api/retrieve-a-pipeline) in the `deployment_runtime_status` field:
```

Comment:
for new users there are no "new" or "old" runtime states.
The pipeline state transition diagram may involve the following states.

BTW: does some other document need to be updated to reflect these states?